service/internal/security/standard_crypto.go

```diff
@@ -135,16 +198,64 @@ func (s StandardCrypto) RSAPublicKeyAsJSON(keyID string) (string, error) {
 	return string(jsonPublicKey), nil
 }
 
-func (s StandardCrypto) GenerateNanoTDFSymmetricKey([]byte) ([]byte, error) {
-	return nil, errNotImplemented
+func (s StandardCrypto) GenerateNanoTDFSymmetricKey(ephemeralPublicKey []byte) ([]byte, error) {
+	keyLengthBytes := len(ephemeralPublicKey)
+	if keyLengthBytes <= 0 {
+		return nil, errors.New("key length should be positive")
+	}
+	key := make([]byte, keyLengthBytes)
+	_, err := rand.Read(key)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate symmetric key: %w", err)
+	}
+	return key, nil
 }
 
 func (s StandardCrypto) GenerateEphemeralKasKeys() (PrivateKeyEC, []byte, error) {
-	return 0, nil, errNotImplemented
+	// prepare private key template
+	privKey := &ecdsa.PrivateKey{
+		PublicKey: ecdsa.PublicKey{
+			Curve: elliptic.P256(),
+		},
+	}
+	// generate private key
+	privKey, err := ecdsa.GenerateKey(privKey.PublicKey.Curve, rand.Reader)
+	if err != nil {
+		return PrivateKeyEC(0), nil, fmt.Errorf("failed to generate elliptic curve key pair: %w", err)
+	}
+	pubBytes, err := x509.MarshalPKIXPublicKey(&privKey.PublicKey)
+	if err != nil {
+		return PrivateKeyEC(0), nil, fmt.Errorf("failed to marshal private key: %w", err)
+	}
+	return PrivateKeyEC(0), pubBytes, nil
 }
 
-func (s StandardCrypto) GenerateNanoTDFSessionKey(PrivateKeyEC, []byte) ([]byte, error) {
-	return nil, errNotImplemented
+func (s StandardCrypto) GenerateNanoTDFSessionKey(privateKeyHandle PrivateKeyEC, ephemeralPublicKey []byte) ([]byte, error) {
+	// Parse the received elliptic public key
+	pubKeyInterface, err := x509.ParsePKIXPublicKey(ephemeralPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse public key: %w", err)
+	}
+
+	// We have to ensure we have an ECDSA public key
+	pubKey, ok := pubKeyInterface.(*ecdsa.PublicKey)
```

Comment:
This should ECDH not ECDSA

Reply:
Session key = KDF(ECDH(sdk-session-public-key, kas-ephemeral-private-key), 'L1L')

L1L is the salt

Reply:
ocrypto module should have this functionality.